tests/test_allow_none_vs_default.py

```diff
@@ -0,0 +1,191 @@
+        self.assertTrue(suite.valid)
+        self.assertEqual(0, suite.snapshot.my_value)
+
+    def test_allow_none_no_default_not_required(self):
+        schema = {
+            MK.Type: types.NamedDict,
+            MK.Content: {
+                "my_value": {
+                    MK.Type: types.Integer,
+                    MK.AllowNone: True,
+                    MK.Required: False,
+                },
+            },
+        }
+
+        for value in (-1, 4, 1000, None):
+            suite = configsuite.ConfigSuite({"my_value": value}, schema)
+            self.assertTrue(suite.valid)
+            self.assertEqual(value, suite.snapshot.my_value)
+
+        suite = configsuite.ConfigSuite({}, schema)
+        self.assertTrue(suite.valid)
+        self.assertEqual(None, suite.snapshot.my_value)
+
+    def test_disallow_none_default_not_required(self):
+        schema = {
+            MK.Type: types.NamedDict,
+            MK.Content: {
+                "my_value": {
+                    MK.Type: types.Integer,
+                    MK.AllowNone: False,
+                    MK.Default: 0,
+                    MK.Required: False,
+                },
+            },
+        }
+
+        for value in (-1, 4, 1000):
+            suite = configsuite.ConfigSuite({"my_value": value}, schema)
+            self.assertTrue(suite.valid, suite.errors)
+            self.assertEqual(value, suite.snapshot.my_value)
+
+            suite = configsuite.ConfigSuite({}, schema)
+            self.assertTrue(suite.valid)
+            self.assertEqual(0, suite.snapshot.my_value)
+
+        suite = configsuite.ConfigSuite({"my_value": None}, schema)
+        self.assertFalse(suite.valid)
+        self.assertEqual(None, suite.snapshot.my_value)
+
+    def test_disallow_none_no_default_not_required(self):
+        schema = {
+            MK.Type: types.NamedDict,
+            MK.Content: {
+                "my_value": {
+                    MK.Type: types.Integer,
+                    MK.AllowNone: False,
+                    MK.Required: False,
+                },
+            },
+        }
+
+        with self.assertRaises(ValueError) as error_context:
+            configsuite.ConfigSuite({}, schema)
+        self.assertIn(
+            "Non required types must allow None", str(error_context.exception)
+        )
+
+    def test_allow_none_default_required(self):
+        schema = {
+            MK.Type: types.NamedDict,
+            MK.Content: {
+                "my_value": {
+                    MK.Type: types.Integer,
+                    MK.AllowNone: True,
+                    MK.Default: 0,
+                    MK.Required: True,
+                },
+            },
+        }
+
+        with self.assertRaises(ValueError) as error_context:
+            configsuite.ConfigSuite({}, schema)
+        self.assertIn("Required can not have Default", str(error_context.exception))
+
+    def test_allow_none_no_default_required(self):
```

Comment:
Feel this is the same as the `MK.AllowNone` issue, this is testing that `MK.Required` handles a config without a required configuration set.